File: src/java/org/elastos/hive/subscription/payment/Order.java
package org.elastos.hive.subscription.payment;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import org.elastos.hive.subscription.PricingPlan;

public class Order {
	@SerializedName("order_id")
	private String orderId;
	private String subscription;
	@SerializedName("pricing_name")
	private String pricingName;
	@SerializedName("ela_amount")
	private Float elaAmount;
	@SerializedName("ela_address")
	private String elaAddress;
	private String proof;
	@SerializedName("create_time")
	private Integer createTime;

	public String getOrderId() {
		return orderId;
	}

	public void setOrderId(String orderId) {
		this.orderId = orderId;
	}

	public String getSubscription() {
		return subscription;
	}

	public void setSubscription(String subscription) {
		this.subscription = subscription;
	}

	public String getPricingName() {
		return pricingName;
	}

	public void setPricingName(String pricingName) {
		this.pricingName = pricingName;
	}

	public Float getElaAmount() {
		return elaAmount;
	}

	public void setElaAmount(Float elaAmount) {
		this.elaAmount = elaAmount;
	}

	public String getElaAddress() {
		return elaAddress;
	}

	public void setElaAddress(String elaAddress) {
		this.elaAddress = elaAddress;
	}

	public String getProof() {
		return proof;
	}

	public void setProof(String proof) {
		this.proof = proof;
	}

	public Integer getCreateTime() {
		return createTime;
	}

	public void setCreateTime(Integer createTime) {
		this.createTime = createTime;
	}
}

File: src/domain/payment/order.ts
/**
 * The order is used for payment module and represents and order to upgrade the service of the vault or the backup.
 */
export class Order {
	private order_id: string;
	private subscription: string;
	private pricing_name: string;
	private ela_amount: number;
	private ela_address: string;
	private proof: string;
	private create_time: Date;

	public getOrderId(): string {
		return this.order_id;
	}

	public setOrderId(orderId: string): Order {
		this.order_id = orderId;
        return this;
	}

	public getSubscription(): string {
		return this.subscription;
	}

	public setSubscription(subscription: string): Order {
		this.subscription = subscription;
        return this;
	}

	public getPricingName(): string {
		return this.pricing_name;
	}

	public setPricingName(pricingName: string): Order {
		this.pricing_name = pricingName;
        return this;
	}

	public getElaAmount(): number {
		return this.ela_amount;
	}

	public setElaAmount(elaAmount: number): Order {
		this.ela_amount = elaAmount;
        return this;
	}

	public getElaAddress(): string {
		return this.ela_address;
	}

	public setElaAddress(elaAddress: string): Order {
		this.ela_address = elaAddress;
        return this;
	}

	public getProof(): string {
		return this.proof;
	}

	public setProof(proof: string): Order {
		this.proof = proof;
        return this;
	}

	public getCreateTime(): Date {
		return this.create_time;
	}

	public setCreateTime(createTime: Date): Order {
		this.create_time = createTime;
        return this;
	}
}
